src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -361,7 +361,7 @@ public void Save(string path)
             {
                 jsonWriter.Formatting = Formatting.Indented;
 
-                Write(writer, PackageSpecWriter.Write);
+                Write(writer, compressed: false, PackageSpecWriter.Write);
```

Comment:
I wish there were some tests to ensure that the compression is only turned on in the right scenarios :) 
I imagine there are write tests, so maybe we are ok.